Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -119,6 +149,32 @@ endRideTransaction driverId bookingId ride mbFareParams mbRiderDetailsId merchan
     updateDriverWeeklyZscore ride rideDate weekStartDate weekEndDate driverWeeklyZscore ride.chargeableDistance
   DLoc.updateOnRideCacheForCancelledOrEndRide driverId merchantId
   SRide.clearCache $ cast driverId
+  unless (driverRideFee <= 0) $ do
+    maxShards <- asks (.maxShards)
+    isPendingPaymentJobScheduled <- getPendingPaymentCache driverFee.endTime
```

Comment:
here we are trying to make only one job for each window right ? but if window size changes from 1 day to something else, then it will start creating more jobs for the same window as the key expiry time is hardcoded to 86400. can we make the expiry time also `pendingPaymentJobTs`